src/coreclr/src/jit/lsrabuild.cpp

```diff
@@ -314,6 +314,7 @@ void LinearScan::resolveConflictingDefAndUse(Interval* interval, RefPosition* de
                 // This is case #2.  Use the useRegAssignment
                 INDEBUG(dumpLsraAllocationEvent(LSRA_EVENT_DEFUSE_CASE2, interval));
                 defRefPosition->registerAssignment = useRegAssignment;
+                defRefPosition->isFixedRegRef      = useRefPosition->isFixedRegRef;
```

Comment:
I had originally thought that it would be better to always set `isFixedRegRef` when the candidates (`registerAssignment`) are limited to a single register, but that causes correctness issues, because the register allocator thinks that it can avoid adding a `COPY` or `RELOAD` in the case of conflicting requirements on def and use. So I believe that these changes need to be reverted.